src/prefect/cli/flow_run.py

```diff
@@ -57,19 +57,57 @@ async def ls(
     flow_name: List[str] = typer.Option(None, help="Name of the flow"),
     limit: int = typer.Option(15, help="Maximum number of flow runs to list"),
     state: List[str] = typer.Option(None, help="Name of the flow run's state"),
-    state_type: List[StateType] = typer.Option(
-        None, help="Type of the flow run's state"
-    ),
+    state_type: List[str] = typer.Option(None, help="Type of the flow run's state"),
 ):
     """
-    View recent flow runs or flow runs for specific flows
+    View recent flow runs or flow runs for specific flows.
+
+    Arguments:
+
+        flow_name: Name of the flow
+
+        limit: Maximum number of flow runs to list. Defaults to 15.
+
+        state: Name of the flow run's state. Can be provided multiple times. Options are 'SCHEDULED', 'PENDING', 'RUNNING', 'COMPLETED', 'FAILED', 'CRASHED', 'CANCELLING', 'CANCELLED', 'PAUSED', 'SUSPENDED', 'AWAITINGRETRY', 'RETRYING', and 'LATE'.
+
+        state_type: Type of the flow run's state. Can be provided multiple times. Options are 'SCHEDULED', 'PENDING', 'RUNNING', 'COMPLETED', 'FAILED', 'CRASHED', 'CANCELLING', 'CANCELLED', 'CRASHED', and 'PAUSED'.
+
+    Examples:
+
+    $ prefect flow-runs ls --state Running
+
+    $ prefect flow-runs ls --state Running --state late
+
+    $ prefect flow-runs ls --state-type RUNNING
+
+    $ prefect flow-runs ls --state-type RUNNING --state-type FAILED
     """
 
+    # Handling `state` and `state_type` argument validity in the function instead of by specifying
+    # List[StateType] and List[StateName] in the type hints, allows users to provide
+    # case-insensitive arguments for `state` and `state_type`.
+
     state_filter = {}
     if state:
-        state_filter["name"] = {"any_": state}
+        upper_cased_states = [s.upper() for s in state]
+        if not all(s in StateName.__members__ for s in upper_cased_states):
+            exit_with_error(
+                f"Invalid state name. Options are {', '.join([name.value for name in StateName])}."
+            )
```

Comment:
Since users can freely create new state _names_, do we want to be this strict on the name search?  (totally agree on being strict with the _type_ search)

Reply:
Yeah good catch! Since we don't advertise that users can create new state names, it has slipped my mind. 

I've applied changes in [5ba6563](https://github.com/PrefectHQ/prefect/pull/13152/commits/5ba6563473175f9288506542a72a13b854743627) and updated the test in [3315e5b](https://github.com/PrefectHQ/prefect/pull/13152/commits/3315e5bc464f032ec09dd0d3e3e591cb981d470b)